### ISSUE TITLE
sr_http    https://username@destination    is used as is.

### ISSUE THREAD

**Issue by michelgrenier** [MetPX/sarracenia#212]
With the japan poll I came into this situation:

sr_poll  posts messages with the username in the message.baseurl
when it is found in the credentials. It is a default behavior.
It needs to put the user since  there can be many users
for the same  server 

The problem encountered is that sr_http tries to download the
file with  username@  into the url ... and that did not worked

THE FIX ...  in sr_http.py  trim or rebuild the URL to use at download
make sure not to have  username@    or  username:password@  

The easy way around for now was to use this option in sr_poll

post_base_url  https://remoteserver 

There is no case of many users on an https remote server for now.

**Comment by petersilva** [comment]
Why is the user in the base_url at all?  The user should affect how to authenticate... it doesn't add anything to the location.  To me, the solution here is that post should not be posting with a username.

**Comment by michelgrenier** [comment]
No no… unfortunately it is important.
There are feed, poll or whatever  aiming at the same server
ex. pxpaz, joule (or imap)… etc

aiming at the right account is important.

MG

**Comment by petersilva** [comment]
I don't believe you on this one... but you are now a pumpkin, will need to discuss with others... see if anyone else understands.

**Comment by michelgrenier** [comment]
Pompkin say there are different accounts on pxpaz and joule for sure. Dont remember if sftp or ftp.  If there are proper permissions then yes any account could get their products... from absolute path.... good continuation 

Envoyé de mon iPhone
